spec/project-spec.coffee

```diff
@@ -13,6 +13,37 @@ describe "Project", ->
   beforeEach ->
     atom.project.setPaths([atom.project.getDirectories()[0]?.resolve('dir')])
 
+  describe "constructor", ->
+    it "tries to update repositories when a new RepositoryProvider is registered", ->
+      atom.project.setPaths(["/tmp"])
+      expect(atom.project.getRepositories()).toEqual [null]
+      expect(atom.project.repositoryProviders.length).toEqual 1
+
+      # Register a new RepositoryProvider.
+      dummyRepository = destroy: () ->
+      repositoryProvider =
+        repositoryForDirectory: (directory) -> Promise.resolve(dummyRepository)
+        repositoryForDirectorySync: (directory) -> dummyRepository
+      atom.packages.serviceHub.provide(
+        "atom.repository-provider", "0.1.0", repositoryProvider)
+
+      expect(atom.project.repositoryProviders.length).toBe 2
+      expect(atom.project.getRepositories()).toEqual [dummyRepository]
+
+    it "does not update @repositories if every path has a Repository", ->
```

Comment:
Yep, exactly. /tmp does not have a repository initially and then the
provider supplies one.

On Friday, February 13, 2015, Max Brunsfeld <notifications@github.com>
wrote:

> In spec/project-spec.coffee
> <https://github.com/atom/atom/pull/5537#discussion_r24712693>:
>
> > +      atom.project.setPaths(["/tmp"])
> > +      expect(atom.project.getRepositories()).toEqual [null]
> > +      expect(atom.project.repositoryProviders.length).toEqual 1
> > +
> > +      # Register a new RepositoryProvider.
> > +      dummyRepository = destroy: () ->
> > +      repositoryProvider =
> > +        repositoryForDirectory: (directory) -> Promise.resolve(dummyRepository)
> > +        repositoryForDirectorySync: (directory) -> dummyRepository
> > +      atom.packages.serviceHub.provide(
> > +        "atom.repository-provider", "0.1.0", repositoryProvider)
> > +
> > +      expect(atom.project.repositoryProviders.length).toBe 2
> > +      expect(atom.project.getRepositories()).toEqual [dummyRepository]
> > +
> > +    it "does not update @repositories if every path has a Repository", ->
>
> Is it true that every path has a repository in this case? It looks like
> the path is being set to spec/fixtures/dir here, which doesn't contain a
> git repo.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/atom/atom/pull/5537/files#r24712693>.
>